vtex/styleguide
# Updates versions of eslint configuration packages

#### What is the purpose of this pull request?
Updates the `eslint-config-vtex-react` to the latest version and remove some unnecessary dependencies and eslint rules due to the upgrade.

#### What problem is this solving?
Some useful warning and rules (e.g. `react-hooks/exhaustive-deps`) weren't properly setup with the old configuration, causing some bugs that were hard to track down. With these updates we can now easily see which dependencies in our effects/callbacks/memos are missing and have complete support for typescript files :tada:.

#### How should this be manually tested?
Run `yarn lint` in your terminal, and there should be no errors. If you think of a component that has dependencies issues in their effects you can open them and see which are the errors.

#### Types of changes
